Feature/multi card

kunroku opened the pull request.

yu23ki14 committed:
Merge pull request #131 from senspace-studio/staging

Fix: change amount of degen

yu23ki14 committed:
Merge pull request #132 from senspace-studio/staging

Release: I want you joker meme

yu23ki14 committed:
Merge pull request #133 from senspace-studio/staging

Hotfix: change desc

yu23ki14 committed:
change share

yu23ki14 committed:
tmp

yu23ki14 committed:
tmp

yu23ki14 committed:
tmp

yu23ki14 committed:
Merge pull request #135 from senspace-studio/staging

Release: change multiplier

yu23ki14 committed:
Merge pull request #138 from senspace-studio/staging

Bugfix: farcaster username

yu23ki14 committed:
Merge branch 'staging'

yu23ki14 committed:
Merge pull request #141 from senspace-studio/staging

Release: Door Open

yu23ki14 committed:
deploy

yu23ki14 committed:
Merge pull request #143 from senspace-studio/staging

Release: Stats

yu23ki14 committed:
Merge pull request #146 from senspace-studio/staging

Release: Update contract

yu23ki14 committed:
contract base

yu23ki14 committed:
tmp

yu23ki14 committed:
multicard smartcontract

Commit:
backend Multi Card 対応

Commit:
Merge branch 'feature/multi-card' of https://github.com/senspace-studio/card into feature/multi-card

Commit:
backend Multi Card 対応 修正

Commit:
backend Multi Card 対応 修正

yu23ki14 commented on backend/server/src/modules/war/war.controller.ts:
```diff
@@ -51,7 +44,11 @@ export class WarController {
-  async getAllReservedGames(@Param('orderBy') orderBy: 'ASC' | 'DESC') {
+  async getAllReservedGames(
+    @Param('orderBy') orderBy: 'ASC' | 'DESC',
+    @Query('card_length') card_length: string,
```
こちらcard_lengthのロジックは未実装ですかね？

kunroku replied:
確認ありがとうございます。こちら修正入れてpushしました。

yu23ki14 reviewed (commented):
おおむね問題なさそうです、ありがとうございます。1点確認いれました。

Commit:
getAllReservedGames fix hand_length

yu23ki14 commented on backend/server/src/modules/war/war.controller.ts:
```diff
@@ -51,8 +44,15 @@ export class WarController {
-  async getAllReservedGames(@Param('orderBy') orderBy: 'ASC' | 'DESC') {
-    const games = await this.warService.getAllReservedGames(orderBy || 'ASC');
+  async getAllReservedGames(
+    @Param('orderBy') orderBy: 'ASC' | 'DESC',
```
ここ@Paramになっているのが、@Queryですかね？今回のPRに関係ないかもですが修正いただけると！

yu23ki14 reviewed (commented):
小さい修正指示いれました！
他は問題なさそうです！

yu23ki14 committed:
fix typo

barubora3 committed:
multi用の素材を登録

barubora3 committed:
multi card用にframeを更新

yu23ki14 committed:
tmp

barubora3 committed:
コントラクトからの使用カード取得処理をマルチ用に修正

barubora3 committed:
コントラクトからの使用カード取得処理をマルチ用に修正

barubora3 committed:
Merge branch 'feature/multi_card_frame' of https://github.com/senspace-studio/card into feature/multi_card_frame

# Conflicts:
#	frame/api/war.tsx

barubora3 commented on backend/server/src/modules/war/war.service.ts:
```diff
@@ -388,6 +419,47 @@ export class WarService {
+    const sumOfCard = tokenIdList.reduce((a, b) => a + (b === 14 ? 0 : b));
+    if (sumOfCard < 14 || 25 < sumOfCard) {
+      throw new Error('invalid card sum');
+    }
```
ここの合計値のチェックですが、1枚の時はスキップした方が良さそうです

Commit:
手札枚数チェック修正

yu23ki14 committed:
tmp

yu23ki14 committed:
Merge branch 'feature/multi-card' of github.com:senspace-studio/card into feature/multi-card

Commit:
手札合計の計算方法を修正

barubora3 committed:
カード1枚時のエラー修正
マルチでの合計値のチェック処理の修正

yu23ki14 committed:
Merge branch 'feature/multi-card' of github.com:senspace-studio/card into feature/multi-card

yu23ki14 committed:
fix conflict

yu23ki14 committed:
fix conflict

yu23ki14 committed:
Merge pull request #149 from senspace-studio/feature/multi_card_frame

Feature/multi card frame

yu23ki14 committed:
ssh

yu23ki14 committed:
fix frog version

yu23ki14 committed:
card array

yu23ki14 committed:
menu

barubora3 committed:
sharpのパラメータが少数になりエラーが発生するのを修正

yu23ki14 committed:
select game mode

yu23ki14 committed:
fix conflict

yu23ki14 committed:
Merge pull request #153 from senspace-studio/feature/multi-card-frame-fix

sharpのパラメータが少数になりエラーが発生するのを修正

yu23ki14 committed:
update stack multiplier for multi card

yu23ki14 committed:
frame rewards link

yu23ki14 committed:
Merge pull request #158 from senspace-studio/feature/multi-card-stack

update stack multiplier for multi card

yu23ki14 committed:
pick card intro

barubora3 committed:
バグフィックス
- select-game-modeを経由してゲームスタートするように
- 上記に付随して選択したカード数がselect-game-modeの値と一致するかチェック
- カード選択フレームにDrawボタンを追加
- エラーメッセージ変更
- Jokerを2枚以上使用していないかのチェックを追加
- JokerKillingでDrawになった時のルートを追加
- シェア時に /result のframeのcard,c_cardが文字列扱いになっていたためArrayにコンバートする処理を追加

barubora3 committed:
Merge branch 'feature/multi-card-frame-fix' of https://github.com/senspace-studio/card into feature/multi-card-frame-fix

yu23ki14 committed:
change draw multiplier

barubora3 committed:
カード選択時に何枚pickすべきかを表示

barubora3 committed:
Joker二枚出しの時のメッセージを修正

yu23ki14 committed:
fix conflict

yu23ki14 committed:
Merge pull request #160 from senspace-studio/feature/multi-card-frame-fix

Fix: Mulit War Frame Bugfix

yu23ki14 committed:
cache control

barubora3 committed:
actionからmakegameする時に最初に枚数を選ぶように

yu23ki14 committed:
Merge pull request #162 from senspace-studio/feature/multi-card-action-fix

actionからmakegameする時に最初に枚数を選ぶように

yu23ki14 committed:
remove invitation check

yu23ki14 committed:
fix war
